pcicrawler/cli.py

```diff
@@ -65,6 +74,283 @@ def jsonify(dev, hexify=False, vpd=False, aer=False):
+        def __init__(self,
+                     device: PCIDevice,
+                     expected_results: str,
+                     ocp_run: tv.TestRun):
+            self._test_name = "pci_device_id_check"
+            super().__init__(device, expected_results, ocp_run)
+
+        def run(self):
+            return super().run(self._device.device_id)
+
+
+    class check_class_id(OCPTestStep):
+        def __init__(self,
+                     device: PCIDevice,
+                     expected_results: str,
+                     ocp_run: tv.TestRun):
+            self._test_name = "pci_class_id_check"
+            super().__init__(device, expected_results, ocp_run)
+
+        def run(self):
+            return super().run(self._device.class_id)
+
+
+    class check_physical_slot(OCPTestStep):
+        def __init__(self,
+                     device: PCIDevice,
+                     expected_results: str,
+                     ocp_run: tv.TestRun):
+            self._test_name = "pci_physical_slot_check"
+            super().__init__(device, expected_results, ocp_run)
+
+        def run(self):
+            return super().run(self._device.express_slot.slot)
+
+
+    class check_current_link_speed(OCPTestStep):
+        def __init__(self,
+                     device: PCIDevice,
+                     expected_results: str,
+                     ocp_run: tv.TestRun):
+            self._test_name = "pci_current_link_speed_check"
+            super().__init__(device, expected_results, ocp_run)
+
+        def run(self):
+            return super().run(self._device.express_link.cur_speed)
+
+
+    class check_current_link_width(OCPTestStep):
+        def __init__(self,
+                     device: PCIDevice,
+                     expected_results: str,
+                     ocp_run: tv.TestRun):
+            self._test_name = "pci_current_link_width_check"
+            super().__init__(device, expected_results, ocp_run)
+
+        def run(self):
+            return super().run(self._device.express_link.cur_width)
+
+
+    class check_capable_link_speed(OCPTestStep):
+        def __init__(self,
+                     device: PCIDevice,
+                     expected_results: str,
+                     ocp_run: tv.TestRun):
+            self._test_name = "pci_capable_link_speed_check"
+            super().__init__(device, expected_results, ocp_run)
+        
+        def run(self):
+            return super().run(self._device.express_link.capable_speed)
+
+
+    class check_capable_link_width(OCPTestStep):
+        def __init__(self,
+                     device: PCIDevice,
+                     expected_results: str,
+                     ocp_run: tv.TestRun):
+            self._test_name = "pci_capable_link_width_check"
+            super().__init__(device, expected_results, ocp_run)
+        
+        def run(self):
+            return super().run(self._device.express_link.capable_width)
+
+
+    class check_address(OCPTestStep):
```

Comment:
all of this logic should definitely not be in `cli.py`. these are diagnosis related, not CLI

Reply:
moviing all the logic to a separate python file